### PR TITLE
consist conceptual and landing data build

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. define conceptual model, used by building conceptual and landing data
2. null/empty array value's key should be removed from output

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4830)

### CODE REVIEW

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -212,19 +217,17 @@ private static (List<Error> errors, JObject model, InputMetadata inputMetadata)
...
+            var (schemaTransformError, transformedToken) = schemaTemplate.JsonSchemaTransformer.TransformContent(file, context, obj);
             errors.AddRange(schemaTransformError);
 
             var (metaErrors, inputMetadata) = context.MetadataProvider.GetMetadata(file);
             errors.AddRange(metaErrors);
 
-            var pageMetadata = new OutputMetadata();
-            var pageModel = transformedToken;
-
+            var pageModel = transformedToken as JObject;
```
Comment (live1206) [inline src/docfx/build/page/BuildPage.cs]: where shall we add the type validation and throw `UnexpectedType`?

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -384,7 +384,7 @@ public static Error ViolateSchema(SourceInfo source, string message)
         /// The input value type does not match expected value type.
         /// </summary>
         /// Behavior: ✔️ Message: ❌
-        public static Error UnexpectedType(SourceInfo source, string expectedType, string actualType)
+        public static Error UnexpectedType(SourceInfo source, object expectedType, object actualType)
```
Comment (live1206) [inline src/docfx/Errors.cs]: why not do a `ToString()` from the caller to avoid the implicit cast?

File: src/docfx/build/page/BuildPage.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/build/page/BuildPage.cs]: I add it at the beginning of LoadSchemaDocument

File: src/docfx/build/xref/InternalXrefMapBuilder.cs [hunk 3]
```diff
@@ -98,8 +98,13 @@ private static (Error error, InternalXrefSpec spec, Document doc) LoadMarkdown(C
             return (error, xref, file);
         }
 
-        private static (List<Error> errors, IReadOnlyList<InternalXrefSpec> specs) LoadSchemaDocument(Context context, JObject obj, Document file)
+        private static (List<Error> errors, IReadOnlyList<InternalXrefSpec> specs) LoadSchemaDocument(Context context, JToken token, Document file)
         {
+            if (!(token is JObject obj))
```
Comment (yufeih) [inline src/docfx/build/xref/InternalXrefMapBuilder.cs]: this check here is not needed

**Comment by OsmondJiang** [comment]
The diff are expected.

**Pull request merged**